test/versions/e2e.spec.ts

```diff
@@ -135,18 +153,85 @@ describe('versions', () => {
       await expect(page.locator('.row-2 .cell-_status')).toContainText('Draft')
     })
 
-    test('collection - displays proper versions pagination', async () => {
+    test('collection - has versions tab', async () => {
+      await goToDoc(page)
+      const docURL = page.url()
+      const pathname = new URL(docURL).pathname
+
+      const versionsTab = page.locator('.doc-tab', {
+        hasText: 'Versions',
+      })
+
+      expect(versionsTab).toBeTruthy()
+      const href = await versionsTab.locator('a').first().getAttribute('href')
+      expect(href).toBe(`${pathname}/versions`)
+    })
+
+    test('collection - tab displays proper number of versions', async () => {
       await page.goto(url.create)
+      console.log(url.create)
 
       // save a version and check count
       await page.locator('#field-title').fill('title')
       await page.locator('#field-description').fill('description')
       await saveDocAndAssert(page)
 
-      await page.goto(`${page.url()}/versions`)
+      const versionsTab = page.locator('.doc-tab', {
+        hasText: 'Versions',
+      })
+
+      const versionCount = await versionsTab.locator('.doc-tab__count').first().textContent()
+      expect(versionCount).toBe('1')
+
+      // save another version and check count again
+      await page.locator('#field-title').fill('title edit')
+      await saveDocAndAssert(page)
+
+      await wait(100) // wait for save and rerender
+      const versionCount2 = await versionsTab.locator('.doc-tab__count').first().textContent()
+      expect(versionCount2).toBe('2')
+
+      const make10More = async () => {
+        for (let i = 0; i < 9; i++) {
+          await wait(50) // wait for save and rerender
+          await page.locator('#field-title').fill(`title ${i + 1}`)
+          await saveDocAndAssert(page)
+        }
+      }
+
+      await make10More()
```

Comment:
You should just use the payload local API in the e2e to insert more data.
Alternatively I don't see a ton of value in this e2e since it is covered by the int tests. If you have some e2e checks it should be minimal to verify that it is showing the values on the page without necessarily needing to do so much work in the browser for setup.

Reply:
Well the original issue was specifically noticed because the tab was not displaying the proper count. So I wanted to make sure I specific covered their scenario.

Reply:
I can move this to the local API, that is a good idea.

Reply:
This is done, I've also cleaned up this PR to remove all the unrelated changes (moved those to https://github.com/payloadcms/payload/pull/3908).